File: CH13/RaiseMan/PreferenceController.rb
# PreferenceController.rb
# RaiseMan
#
# Created by daniellopes on 04/05/10.
# Copyright 2010 area criacoes. All rights reserved.


class PreferenceController < NSWindowController
  attr_accessor :colorWell, :checkbox
	
  BNRTableBgColorKey = "TableBackgroundColor"
  BNREmptyDocKey = "EmptyDocumentFlag"	
	
	def init
    initWithWindowNibName("Preferences")
    self
	end
	
	def windowDidLoad
		@colorWell.setColor(tableBgColor)
		@checkbox.setState(emptyDoc)
	end
	
	def changeBackgroundColor(sender)
		color = @colorWell.color
		colorAsData = NSKeyedArchiver.archivedDataWithRootObject(color)
		defaults = NSUserDefaults.standardUserDefaults
		defaults.setObject(colorAsData, forKey:BNRTableBgColorKey)
	end
	
	def changeNewEmptyDoc(sender)
    state = @checkbox.state
    state = ( state == 1 ) ? true : false
		
		defaults = NSUserDefaults.standardUserDefaults
		defaults.setBool(state, forKey:BNREmptyDocKey)
	end	

	def	tableBgColor
		defaults = NSUserDefaults.standardUserDefaults
		colorAsData = defaults.objectForKey(BNRTableBgColorKey)
		NSKeyedUnarchiver.unarchiveObjectWithData(colorAsData)
	end
	
	def emptyDoc
		defaults = NSUserDefaults.standardUserDefaults
		defaults.boolForKey(BNREmptyDocKey)	
	end

end

File: CH13/RaiseMan/PreferenceController.m
//
//  PreferenceController.m
//  RaiseMan
//
//  Created by Aaron Hillegass on 10/15/07.
//  Copyright 2007 __MyCompanyName__. All rights reserved.
//

#import "PreferenceController.h"

NSString *BNRTableBgColorKey = @"TableBackgroundColor";
NSString *BNREmptyDocKey = @"EmptyDocumentFlag";


@implementation PreferenceController

- (id)init
{
	if (![super initWithWindowNibName:@"Preferences"]) 
		return nil;
	
	return self;
}
- (NSColor *)tableBgColor
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	NSData *colorAsData = [defaults objectForKey:BNRTableBgColorKey];
	return [NSKeyedUnarchiver unarchiveObjectWithData:colorAsData];
}
- (BOOL)emptyDoc
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	return [defaults boolForKey:BNREmptyDocKey];
}

- (void)windowDidLoad
{
	[colorWell setColor:[self tableBgColor]];
	[checkbox setState:[self emptyDoc]];
}

- (IBAction)changeBackgroundColor:(id)sender
{
	NSColor *color = [colorWell color];
	NSData *colorAsData;
	colorAsData = [NSKeyedArchiver archivedDataWithRootObject:color];
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	[defaults setObject:colorAsData forKey:BNRTableBgColorKey];
}

- (IBAction)changeNewEmptyDoc:(id)sender
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	[defaults setBool:[checkbox state] forKey:BNREmptyDocKey];
}

- (IBAction)resetFactoryDefaults:(id)sender
{
	[NSUserDefaults resetStandardUserDefaults];
	[colorWell setColor:[self tableBgColor]];
	[checkbox setState:[self emptyDoc]];	
}

@end
